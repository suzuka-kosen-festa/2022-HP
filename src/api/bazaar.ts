import type { Bazaar, BazaarPrices, BazaarType } from "./prisma";

export type BazaarWithoutId = Omit<Bazaar, "id"> & {
  prices: ReadonlyArray<BazaarPrices>;
};

export type BazaarWithId = Bazaar & {
  prices: ReadonlyArray<BazaarPrices>;
};
