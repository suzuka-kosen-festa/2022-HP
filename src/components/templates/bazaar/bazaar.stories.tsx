import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { ComponentPropsWithoutRef } from "react";
import { Bazaar } from ".";

type T = typeof Bazaar;
type Story = ComponentStoryObj<T>;
type Meta = ComponentMeta<T>;

const args: ComponentPropsWithoutRef<T> = {
  bazaars: [
    {
      id: 1,
      name: "A.作品展示",
      descriptions: "美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！",
      image: "https://panproduct.com/blog/wp-content/uploads/2021/07/32.png",
      prices: [{ id: 1, price: "無料", bazaarId: 1 }],
      group: "美術部",
      group_type: "recreation",
    },
    {
      id: 1,
      name: "A.作品展示",
      descriptions: "美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！",
      image: "https://panproduct.com/blog/wp-content/uploads/2021/07/32.png",
      prices: [{ id: 1, price: "無料", bazaarId: 1 }],
      group: "美術部",
      group_type: "recreation",
    },
    {
      id: 1,
      name: "A.作品展示",
      descriptions: "美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！",
      image: "https://panproduct.com/blog/wp-content/uploads/2021/07/32.png",
      prices: [{ id: 1, price: "無料", bazaarId: 1 }],
      group: "美術部",
      group_type: "recreation",
    },
    {
      id: 1,
      name: "A.作品展示",
      descriptions: "美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！",
      image: "https://panproduct.com/blog/wp-content/uploads/2021/07/32.png",
      prices: [{ id: 1, price: "無料", bazaarId: 1 }],
      group: "美術部",
      group_type: "recreation",
    },
    {
      id: 1,
      name: "A.作品展示",
      descriptions: "美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！",
      image: "https://panproduct.com/blog/wp-content/uploads/2021/07/32.png",
      prices: [{ id: 1, price: "無料", bazaarId: 1 }],
      group: "美術部",
      group_type: "recreation",
    },
    {
      id: 1,
      name: "A.作品展示",
      descriptions: "美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！",
      image: "https://panproduct.com/blog/wp-content/uploads/2021/07/32.png",
      prices: [{ id: 1, price: "無料", bazaarId: 1 }],
      group: "美術部",
      group_type: "recreation",
    },
    {
      id: 1,
      name: "A.作品展示",
      descriptions: "美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！",
      image: "https://panproduct.com/blog/wp-content/uploads/2021/07/32.png",
      prices: [{ id: 1, price: "無料", bazaarId: 1 }],
      group: "美術部",
      group_type: "recreation",
    },
  ],
};

export default {
  args,
  argTypes: {
    bazaars: {
      description: "Bazaars data",
      control: {
        type: "array",
      },
    },
  },
  component: Bazaar,
} as Meta;

export const Default: Story = {};
