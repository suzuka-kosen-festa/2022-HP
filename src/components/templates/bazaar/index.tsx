import type { FC } from "react";
import tw from "twin.macro";
import { Link } from "wouter";
import { useMemo } from "react";
import type { BazaarProperties } from "./types/model";
import { Title } from "../../molecules/title";
import { Switching } from "../../atoms/switching";
import { BazaarCard } from "../../molecules/bazaar-card";
import { Button } from "../../atoms/button";
import { Text } from "../../atoms/text";

const Main = tw.main`pt-1 pb-8 sm:(pt-6 pb-40)`;

const BazaarPageBox = tw.div`flex flex-col justify-center items-center`;

const BazaarCardBox = tw.ul`grid gap-6 grid-cols-1 pt-6 mb-8 xl:(grid-cols-2 pt-32 mb-36) 3xl:(grid-cols-3 pt-48 mb-40)`;

const buttonData = [
  {
    buttonText: "飲食団体",
    query: "eating",
  },
  {
    buttonText: "レク・展示団体",
    query: "recreation",
  },
];

const Bazaar: FC<BazaarProperties> = ({ bazaars }) => {
  const query = new URLSearchParams(window.location.search);
  const isRecreation = query.get("select") === "recreation";
  return (
    <Main aria-label="Bazaar page">
      <Title colorKey="green">Bazaar</Title>
      <BazaarPageBox>
        <Switching buttonData={buttonData} color="green" />
        <BazaarCardBox>
          {bazaars.map(bazaar => (
            <BazaarCard bazaar={bazaar} key={bazaar.id} />
          ))}
        </BazaarCardBox>
        {isRecreation ? (
          <Text css={tw`mb-8 md:mb-36 lg:mb-40`}>
            これらの団体のほかに、C.専攻科・E.化学マジック・S.エコカーも展示を行っています！ぜひ見に行ってみてください！
          </Text>
        ) : null}
        <Link to="/">
          <Button textStyles={tw`text-white`} colorKey="green" as="a" role="button">
            トップページに戻る
          </Button>
        </Link>
      </BazaarPageBox>
    </Main>
  );
};

export { Bazaar };
