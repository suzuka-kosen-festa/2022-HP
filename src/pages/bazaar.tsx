import type { FC } from "react";
import useSWR from "swr";
import { Redirect } from "wouter";
import { Bazaar } from "../components/templates/bazaar";
import { fetcher } from "../libs/axios";
import type { BazaarWithId } from "../api/bazaar";

const BazaarPage: FC = () => {
  const query = new URLSearchParams(window.location.search);
  const isRecreation = query.get("select") === "recreation";
  const { data, error } = useSWR<ReadonlyArray<BazaarWithId>>(
    isRecreation ? "/bazaar/data/recreation" : "/bazaar/data/eating",
    fetcher,
  );
  if (!data) return <p>Loading...</p>;
  if (error) return <Redirect to="/error" />;
  return <Bazaar bazaars={data} />;
};

export { BazaarPage };
