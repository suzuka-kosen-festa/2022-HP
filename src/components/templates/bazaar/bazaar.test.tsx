import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./bazaar.stories";

const { Default } = composeStories(stories);

const options = {
  name: "Bazaar page",
};

describe("(components) templates/bazaar", () => {
  test("to be templates", () => {
    const { container } = render(<Default />);
    expect(container).toBeTemplate();
  });
  test("to be [role=main]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("main", options)).toBeInTheDocument();
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
