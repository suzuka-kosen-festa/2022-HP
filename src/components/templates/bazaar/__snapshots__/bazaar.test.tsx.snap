// Vitest Snapshot v1

exports[`(components) templates/bazaar > take snap shot 1`] = `
<div>
  <main
    aria-label="Bazaar page"
    class="css-10jk6q1"
  >
    <div
      class="css-79elbk"
    >
      <img
        alt="ink"
        class="css-mhf7w5-Title"
        height="580"
        loading="eager"
        src="/statics/Ink3_green.webp"
        width="592"
      />
      <h1
        class="css-1ueto1"
      >
        Bazaar
      </h1>
    </div>
    <div
      class="css-wkpw2c"
    >
      <div
        class="css-1qq4q19"
      >
        <button
          class="css-1w2nvi1-SwitchingButton"
          tabindex="0"
        >
          <p
            class="css-1bpm81q-SwitchingButton"
          >
            飲食団体
          </p>
        </button>
        <button
          class="css-1w2nvi1-SwitchingButton"
          tabindex="0"
        >
          <p
            class="css-1bpm81q-SwitchingButton"
          >
            レク・展示団体
          </p>
        </button>
      </div>
      <ul
        class="css-5z47ei"
      >
        <li
          class="css-4chdo6"
        >
          <img
            alt="バザーに関するイメージ画像"
            class="css-61q9gc-BazaarCard"
            height="300"
            loading="lazy"
            src="https://panproduct.com/blog/wp-content/uploads/2021/07/32.png"
            width="450"
          />
          <div
            class="css-z0erpm"
          >
            <div
              class="css-1q8fkzn"
            >
              <p
                class="css-jprqak"
              >
                美術部
              </p>
            </div>
            <h2
              class="css-hh2gwj"
            >
              A.作品展示
            </h2>
            <p
              class="css-vaqku0-BazaarCard"
            >
              美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！
            </p>
            <ul
              class="css-luxo22"
            >
              <li>
                <p
                  class="css-4gn9jk"
                >
                  無料
                </p>
              </li>
            </ul>
          </div>
        </li>
        <li
          class="css-4chdo6"
        >
          <img
            alt="バザーに関するイメージ画像"
            class="css-61q9gc-BazaarCard"
            height="300"
            loading="lazy"
            src="https://panproduct.com/blog/wp-content/uploads/2021/07/32.png"
            width="450"
          />
          <div
            class="css-z0erpm"
          >
            <div
              class="css-1q8fkzn"
            >
              <p
                class="css-jprqak"
              >
                美術部
              </p>
            </div>
            <h2
              class="css-hh2gwj"
            >
              A.作品展示
            </h2>
            <p
              class="css-vaqku0-BazaarCard"
            >
              美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！
            </p>
            <ul
              class="css-luxo22"
            >
              <li>
                <p
                  class="css-4gn9jk"
                >
                  無料
                </p>
              </li>
            </ul>
          </div>
        </li>
        <li
          class="css-4chdo6"
        >
          <img
            alt="バザーに関するイメージ画像"
            class="css-61q9gc-BazaarCard"
            height="300"
            loading="lazy"
            src="https://panproduct.com/blog/wp-content/uploads/2021/07/32.png"
            width="450"
          />
          <div
            class="css-z0erpm"
          >
            <div
              class="css-1q8fkzn"
            >
              <p
                class="css-jprqak"
              >
                美術部
              </p>
            </div>
            <h2
              class="css-hh2gwj"
            >
              A.作品展示
            </h2>
            <p
              class="css-vaqku0-BazaarCard"
            >
              美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！
            </p>
            <ul
              class="css-luxo22"
            >
              <li>
                <p
                  class="css-4gn9jk"
                >
                  無料
                </p>
              </li>
            </ul>
          </div>
        </li>
        <li
          class="css-4chdo6"
        >
          <img
            alt="バザーに関するイメージ画像"
            class="css-61q9gc-BazaarCard"
            height="300"
            loading="lazy"
            src="https://panproduct.com/blog/wp-content/uploads/2021/07/32.png"
            width="450"
          />
          <div
            class="css-z0erpm"
          >
            <div
              class="css-1q8fkzn"
            >
              <p
                class="css-jprqak"
              >
                美術部
              </p>
            </div>
            <h2
              class="css-hh2gwj"
            >
              A.作品展示
            </h2>
            <p
              class="css-vaqku0-BazaarCard"
            >
              美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！
            </p>
            <ul
              class="css-luxo22"
            >
              <li>
                <p
                  class="css-4gn9jk"
                >
                  無料
                </p>
              </li>
            </ul>
          </div>
        </li>
        <li
          class="css-4chdo6"
        >
          <img
            alt="バザーに関するイメージ画像"
            class="css-61q9gc-BazaarCard"
            height="300"
            loading="lazy"
            src="https://panproduct.com/blog/wp-content/uploads/2021/07/32.png"
            width="450"
          />
          <div
            class="css-z0erpm"
          >
            <div
              class="css-1q8fkzn"
            >
              <p
                class="css-jprqak"
              >
                美術部
              </p>
            </div>
            <h2
              class="css-hh2gwj"
            >
              A.作品展示
            </h2>
            <p
              class="css-vaqku0-BazaarCard"
            >
              美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！
            </p>
            <ul
              class="css-luxo22"
            >
              <li>
                <p
                  class="css-4gn9jk"
                >
                  無料
                </p>
              </li>
            </ul>
          </div>
        </li>
        <li
          class="css-4chdo6"
        >
          <img
            alt="バザーに関するイメージ画像"
            class="css-61q9gc-BazaarCard"
            height="300"
            loading="lazy"
            src="https://panproduct.com/blog/wp-content/uploads/2021/07/32.png"
            width="450"
          />
          <div
            class="css-z0erpm"
          >
            <div
              class="css-1q8fkzn"
            >
              <p
                class="css-jprqak"
              >
                美術部
              </p>
            </div>
            <h2
              class="css-hh2gwj"
            >
              A.作品展示
            </h2>
            <p
              class="css-vaqku0-BazaarCard"
            >
              美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！
            </p>
            <ul
              class="css-luxo22"
            >
              <li>
                <p
                  class="css-4gn9jk"
                >
                  無料
                </p>
              </li>
            </ul>
          </div>
        </li>
        <li
          class="css-4chdo6"
        >
          <img
            alt="バザーに関するイメージ画像"
            class="css-61q9gc-BazaarCard"
            height="300"
            loading="lazy"
            src="https://panproduct.com/blog/wp-content/uploads/2021/07/32.png"
            width="450"
          />
          <div
            class="css-z0erpm"
          >
            <div
              class="css-1q8fkzn"
            >
              <p
                class="css-jprqak"
              >
                美術部
              </p>
            </div>
            <h2
              class="css-hh2gwj"
            >
              A.作品展示
            </h2>
            <p
              class="css-vaqku0-BazaarCard"
            >
              美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！
            </p>
            <ul
              class="css-luxo22"
            >
              <li>
                <p
                  class="css-4gn9jk"
                >
                  無料
                </p>
              </li>
            </ul>
          </div>
        </li>
      </ul>
      <a
        class="css-3os8bq-Button"
        href="/"
        role="button"
        tabindex="0"
      >
        <p
          class="css-1t0ga8k-Button"
        >
          トップページに戻る
        </p>
      </a>
    </div>
  </main>
</div>
`;
